network/secondary_endpoint_linux_test.go

```diff
@@ -0,0 +1,264 @@
+	mac, _ := net.ParseMAC("ab:cd:ef:12:34:56")
+	invalidMac, _ := net.ParseMAC("12:34:56:ab:cd:ef")
+
+	tests := []struct {
+		name       string
+		client     *SecondaryEndpointClient
+		epInfo     *EndpointInfo
+		wantErr    bool
+		wantErrMsg string
+	}{
+		{
+			name: "Add endpoints",
+			client: &SecondaryEndpointClient{
+				netlink:        netlink.NewMockNetlink(false, ""),
+				plClient:       platform.NewMockExecClient(false),
+				netUtilsClient: networkutils.NewNetworkUtils(nl, plc),
+				netioshim:      netio.NewMockNetIO(false, 0),
+				ep:             &endpoint{SecondaryInterfaces: make(map[string]*InterfaceInfo)},
+			},
+			epInfo:  &EndpointInfo{MacAddress: mac},
+			wantErr: false,
+		},
+		{
+			name: "Add endpoints invalid mac",
+			client: &SecondaryEndpointClient{
+				netlink:        netlink.NewMockNetlink(false, ""),
+				plClient:       platform.NewMockExecClient(false),
+				netUtilsClient: networkutils.NewNetworkUtils(nl, plc),
+				netioshim:      netio.NewMockNetIO(false, 0),
+				ep:             &endpoint{SecondaryInterfaces: make(map[string]*InterfaceInfo)},
+			},
+			epInfo:     &EndpointInfo{MacAddress: invalidMac},
+			wantErr:    true,
+			wantErrMsg: "SecondaryEndpointClient Error: " + netio.ErrMockNetIOFail.Error() + ": " + invalidMac.String(),
+		},
+		{
+			name: "Add endpoints interface already added",
+			client: &SecondaryEndpointClient{
+				netlink:        netlink.NewMockNetlink(false, ""),
+				plClient:       platform.NewMockExecClient(false),
+				netUtilsClient: networkutils.NewNetworkUtils(nl, plc),
+				netioshim:      netio.NewMockNetIO(false, 0),
+				ep:             &endpoint{SecondaryInterfaces: map[string]*InterfaceInfo{"eth1": {Name: "eth1"}}},
+			},
+			epInfo:     &EndpointInfo{MacAddress: mac},
+			wantErr:    true,
+			wantErrMsg: "SecondaryEndpointClient Error: eth1 already exists",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.client.AddEndpoints(tt.epInfo)
+			if tt.wantErr {
+				require.Error(t, err)
+				require.Contains(t, tt.wantErrMsg, err.Error(), "Expected:%v actual:%v", tt.wantErrMsg, err.Error())
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, tt.client.ep.SecondaryInterfaces["eth1"].MacAddress, tt.epInfo.MacAddress)
+			}
+		})
+	}
+}
+
+func TestSecondaryDeleteEndpoints(t *testing.T) {
+	nl := netlink.NewMockNetlink(false, "")
+	plc := platform.NewMockExecClient(false)
+
+	tests := []struct {
+		name   string
+		client *SecondaryEndpointClient
+		ep     *endpoint
+	}{
+		{
+			name: "Delete endpoint happy path",
+			client: &SecondaryEndpointClient{
+				netlink:        netlink.NewMockNetlink(false, ""),
+				plClient:       platform.NewMockExecClient(false),
+				netUtilsClient: networkutils.NewNetworkUtils(nl, plc),
+				netioshim:      netio.NewMockNetIO(false, 0),
+			},
+			ep: &endpoint{
+				NetworkNameSpace: fmt.Sprintf("/proc/%d/task/%d/ns/net", os.Getpid(), unix.Gettid()),
```

Comment:
should mock namespace api and not use real namespace

Reply:
using mocknamespaceclient now